design/design-doc-template-stacks-experience.md

```diff
@@ -0,0 +1,505 @@
+stack will do something in response to it. In the case of template
+stacks, that usually will look something like rendering a set of yamls
+and applying the rendered output to the cluster.
+
+#### Install
+
+Installation is much the same as installing any stack:
+
+```
+kubectl crossplane stack install myorg/mystack mystack
+```
+
+This installs the stack from the `myorg/mystack` image, using the name `mystack`.
+
+#### Create
+
+Instantiation is also very similar to any stack:
+
+```
+cat > my-instance.yaml <<EOF
+apiVersion: thing.samples.stacks.crossplane.io/v1alpha1
+kind: ThingInstance
+metadata:
+  name: thinginstance-sample
+spec:
+  myfield: myvalue
+EOF
+
+kubectl apply -f my-instance.yaml
+```
+
+The difference is that underneath, the ThingInstance will be used as the
+input for rendering a template.
+
+#### Delete
+
+When the ThingInstance is deleted, the corresponding resources will also
+be deleted. This is the same behavior as in any stack.
+
+```
+cat > my-instance.yaml <<EOF
+apiVersion: thing.samples.stacks.crossplane.io/v1alpha1
+kind: ThingInstance
+metadata:
+  name: thinginstance-sample
+spec:
+  myfield: myvalue
+EOF
+
+kubectl delete -f my-instance.yaml
+```
+
+The instance is also deleted if the stack is uninstalled.
+
+#### Uninstall
+
+Uninstall looks the same as any other stack:
+
+```
+kubectl crossplane stack uninstall mystack
+```
+
+#### Upgrade and update
+
+Upgrading a stack version, and updating an instance of an object managed
+by the stack, are out of scope of this document. However, one could
+imagine that the process of upgrading a stack version would change a
+version number on object instances which go with the stack, and that the
+version number change would underneath cause the template to be rendered
+again and applied to the cluster.
+
+### The stack.yaml
+
+Configuration for the template stack will be in a `stack.yaml` file at
+the build root of the stack in the repository. For most cases, this will
+mean the `stack.yaml` lives at the root of the repository.
+
+#### Specifying how to process render requests
+
+A Stack author must create a `stack.yaml` in order to configure how
+templates are rendered in response to a render request. Here's an
+example, with comments explaining the directives:
+
+```
+cat > stack.yaml <<EOF
+# This field configures which templates are rendered in response
+# to a given object type. An instance of the object type is
+# considered to be a "render request".
+behaviors:
+  crds:
+    # This is a particular CRD which is being configured. When the
+    # controller sees an object of this type, it will do something.
+    wordpressinstance.wordpress.samples.stacks.crossplane.io:
+      # This is a top-level object to namespace hook configurations.
+      # There can be hooks for multiple different types of events.
+      hooks:
+        # Post create is triggered after an instance is created
+        postCreate:
```

Comment:
`post*` implies that this is the hook run after the `create` hook has been run.  I wonder if it will simplify the semantics of this hook to just simply call it `create`?  same with `postUpdate`.